Document Konflux and CVE workflows

dfarrell07 opened the pull request:
Add agent-executable workflows in .agents/ that:
- Add Konflux bundle builds to new branches
- Add Konflux operator builds to new branches
- Fix CVEs
- Fix Konflux CI violations
- Update bundle SHAs

Also includes commit message guidance and CLAUDE.md index.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt reviewed (approved):
I think this is really useful — even ignoring Claude, this helps document project processes, which is great!

dfarrell07 commented:
The CVE workflow one-shots consistently for 0.21 and 0.19: https://github.com/submariner-io/submariner-operator/pull/3707, https://github.com/submariner-io/submariner-operator/pull/3708

dfarrell07 commented:
The Konflux-adding workflow now consistently one-shots getting 0.19 building and passing all EC: https://github.com/submariner-io/submariner-operator/pull/3711

dfarrell07 commented:
Also one-shot 0.17 and 0.18 CVEs: https://github.com/submariner-io/submariner-operator/pull/3715, https://github.com/submariner-io/submariner-operator/pull/3714

dfarrell07 commented:
The Konflux CI fix workflow was able to create: https://github.com/submariner-io/submariner-operator/pull/3718, https://github.com/submariner-io/submariner/pull/3683, https://github.com/submariner-io/lighthouse/pull/1936, https://github.com/submariner-io/shipyard/pull/2178, https://github.com/submariner-io/subctl/pull/1545

dfarrell07 committed:
Document Konflux and CVE workflows

Add agent-executable workflows in .agents/ that:
- Add Konflux bundle builds to new branches
- Add Konflux operator builds to new branches
- Fix CVEs
- Fix Konflux CI violations
- Update bundle SHAs

Also includes commit message guidance and CLAUDE.md index.

signed-off-by: Daniel Farrell <dfarrell@redhat.com>

tpantelis committed:
Merge branch 'devel' into kf_claud
